src/main/java/com/google/cloud/spanner/pgadapter/utils/MutationWriter.java

```diff
@@ -186,39 +195,32 @@ public int writeToSpanner(ConnectionHandler connectionHandler) throws SQLExcepti
     // Reset mutations, mutation counter, and batch size count for a new batch
     this.mutations = new ArrayList<>();
     this.mutationCount = 0;
-    this.batchSize = 0;
     return this.rowCount;
   }
 
-  public void rollback(ConnectionHandler connectionHandler, byte[] payload) throws Exception {
+  public void rollback(ConnectionHandler connectionHandler) throws Exception {
     Connection connection = connectionHandler.getJdbcConnection();
     connection.rollback();
     this.mutations = new ArrayList<>();
     this.mutationCount = 0;
-    this.batchSize = 0;
-    createErrorFile(payload);
+    writeToErrorFile(this.payload.toByteArray());
   }
 
-  public void createErrorFile(byte[] payload) throws IOException {
+  private void createErrorFile() throws IOException {
     File unsuccessfulCopy = new File(ERROR_FILE);
-    if (unsuccessfulCopy.createNewFile()) {
-      this.fileWriter = new FileWriter(ERROR_FILE);
-      writeToErrorFile(payload);
-    } else {
-      System.err.println("File " + unsuccessfulCopy.getName() + " already exists");
-    }
+    this.fileWriter = new FileWriter(unsuccessfulCopy, false);
   }
 
   public void writeToErrorFile(byte[] payload) throws IOException {
-    if (this.fileWriter != null) {
-      this.fileWriter.write(new String(payload, StandardCharsets.UTF_8).trim() + "\n");
+    if (this.fileWriter == null) {
+      createErrorFile();
     }
+    this.fileWriter.write(new String(payload, StandardCharsets.UTF_8).trim() + "\n");
   }
 
   public void writeMutationsToErrorFile() throws IOException {
```

Comment:
add a comment that this is used if Spanner returns an error when we try to commit the mutations

Reply:
Done.

Reply:
actually, I'm thinking about this again and why are we writing the mutations to the file at all? The original intention of the error file was for the user to look through the copy data that they sent to PGAdapter, fix it, and try again. I think we should always be writing the payload data, not the mutation data

Reply:
Changed to write the copy data instead.